kivy/graphics/c_opengl.pxd

```diff
@@ -384,147 +384,148 @@ cdef extern from "gl_redirect.h":
-    cdef void  glTexParameterfv(GLenum target, GLenum pname,  GLfloat* params) nogil
-    cdef void  glTexParameteri(GLenum target, GLenum pname, GLint param) nogil
-    cdef void  glTexParameteriv(GLenum target, GLenum pname,  GLint* params) nogil
-    cdef void  glTexSubImage2D(GLenum target, GLint level, GLint xoffset, GLint yoffset, GLsizei width, GLsizei height, GLenum format, GLenum type,  GLvoid* pixels) nogil
-    cdef void  glUniform1f(GLint location, GLfloat x) nogil
-    cdef void  glUniform1fv(GLint location, GLsizei count,  GLfloat* v) nogil
-    cdef void  glUniform1i(GLint location, GLint x) nogil
-    cdef void  glUniform1iv(GLint location, GLsizei count,  GLint* v) nogil
-    cdef void  glUniform2f(GLint location, GLfloat x, GLfloat y) nogil
-    cdef void  glUniform2fv(GLint location, GLsizei count,  GLfloat* v) nogil
-    cdef void  glUniform2i(GLint location, GLint x, GLint y) nogil
-    cdef void  glUniform2iv(GLint location, GLsizei count,  GLint* v) nogil
-    cdef void  glUniform3f(GLint location, GLfloat x, GLfloat y, GLfloat z) nogil
-    cdef void  glUniform3fv(GLint location, GLsizei count,  GLfloat* v) nogil
-    cdef void  glUniform3i(GLint location, GLint x, GLint y, GLint z) nogil
-    cdef void  glUniform3iv(GLint location, GLsizei count,  GLint* v) nogil
-    cdef void  glUniform4f(GLint location, GLfloat x, GLfloat y, GLfloat z, GLfloat w) nogil
-    cdef void  glUniform4fv(GLint location, GLsizei count,  GLfloat* v) nogil
-    cdef void  glUniform4i(GLint location, GLint x, GLint y, GLint z, GLint w) nogil
-    cdef void  glUniform4iv(GLint location, GLsizei count,  GLint* v) nogil
-    cdef void  glUniformMatrix2fv(GLint location, GLsizei count, GLboolean transpose,  GLfloat* value) nogil
-    cdef void  glUniformMatrix3fv(GLint location, GLsizei count, GLboolean transpose,  GLfloat* value) nogil
-    cdef void  glUniformMatrix4fv(GLint location, GLsizei count, GLboolean transpose,  GLfloat* value) nogil
-    cdef void  glUseProgram(GLuint program) nogil
-    cdef void  glValidateProgram(GLuint program) nogil
-    cdef void  glVertexAttrib1f(GLuint indx, GLfloat x) nogil
-    cdef void  glVertexAttrib1fv(GLuint indx,  GLfloat* values) nogil
-    cdef void  glVertexAttrib2f(GLuint indx, GLfloat x, GLfloat y) nogil
-    cdef void  glVertexAttrib2fv(GLuint indx,  GLfloat* values) nogil
-    cdef void  glVertexAttrib3f(GLuint indx, GLfloat x, GLfloat y, GLfloat z) nogil
-    cdef void  glVertexAttrib3fv(GLuint indx,  GLfloat* values) nogil
-    cdef void  glVertexAttrib4f(GLuint indx, GLfloat x, GLfloat y, GLfloat z, GLfloat w) nogil
-    cdef void  glVertexAttrib4fv(GLuint indx,  GLfloat* values) nogil
-    cdef void  glVertexAttribPointer(GLuint indx, GLint size, GLenum type, GLboolean normalized, GLsizei stride,  GLvoid* ptr) nogil
-    cdef void  glViewport(GLint x, GLint y, GLsizei width, GLsizei height) nogil
+    IF not USE_OPENGL_MOCK:
```

Comment:
Not quite. Even when mocking we still use this file for the constants, which would cause linking issues if not guarded like this. So this file is always included. Otherwise, we'd had to reimport all the constants in the mocking pxd file as well. This seemed cleaner. What do you think?